src/components/Badge/Badge.jsx

```diff
@@ -65,8 +37,31 @@ const Badge = (props) => {
         });
         setTotalBadge(Math.round(count));
       }
-    });
-  }, []);
+    }
+    setOpen(isOpen => !isOpen);
+  };
+
+  const toggleTypes = () => {
+    setOpenTypes(isOpenTypes => !isOpenTypes);
+  };
+
+  useEffect(() => {
+    const userId = props.userId;
+    console.log('This is userId', userId);
+    let count = 0;
+    if (props.userProfile.badgeCollection) {
+      console.log('userProfileBadgeCollectionId', props.userProfile._id);
+      props.userProfile.badgeCollection.forEach(badge => {
+        console.log('badge2', badge);
+        if (badge?.badge?.badgeName === 'Personal Max' || badge?.badge?.type === 'Personal Max') {
+          count += 1;
+        } else {
+          count += badge.count;
+        }
+      });
+      setTotalBadge(Math.round(count));
+    }
+  }, [props.userProfile.badgeCollection, totalBadge]);
```

Comment:
this re-renders the component as userProfile isn't immediately updated after login which is why I am not sure if this is a proper fix. Also, the two state changes re-render the badge components multiple times.